### PR TITLE
Add focus management to the mobile menu open/close interactions

### PR DESCRIPTION
**Pull request by jimmynotjim**
When opening and closing menu states the focus is hidden on the trigger
that moves out of view. This change addresses the issue by moving the
focus to the logical trigger of the panel that comes within view.
There's also a bit of cleanup to avoid setting/unsetting all transitions
on the sibling submenus and narrow the focus (pun intended) to only the
child of the parent trigger.

## Additions

- Added `handleExpandEnd` function to manage the animating state class and focus changes after animations have completed.
- Added logic to only set transitions to the intended child menu.

## Removals

- Removed the logic that set transitions on all child menus.

## Changes

-Updated the expand begin, collpase begin, and collapse end functions to manage the animating state class to better target styles and focus changes to just the child menus.

## Testing

1. `gulp build`
1. `./runserver.sh`
1. In another tab run `lt --port 8000` to start local tunnel
1. Use the mobile menu on both desktop and your local device.

## Screenshots

Open sub menu

screen shot 2018-06-27 at 12 44 39 pm

Close sub menu

screen shot 2018-06-27 at 12 44 32 pm

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Internet Explorer 8, 9, 10, and 11
- [ ] Edge
- [x] iOS Safari
- [ ] Chrome for Android

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [x] JavaScript tests are passing

### CODE REVIEW

File: cfgov/unprocessed/js/organisms/MegaMenuMobile.js [hunk 1]
```diff
@@ -173,21 +169,34 @@ function MegaMenuMobile( menus ) {
...
-       firstMenuLink = _activeMenuDom.querySelector( 'a' );
-       } else {
-       firstMenuLink = _activeMenuDom.querySelector( 'button' );
-       }
-       firstMenuLink.focus(); */
+    menuDom.content.classList.add('u-is-animating');
+  }
+
+  /**
+   * Event handler for when FlyoutMenu expand transition begins.
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/organisms/MegaMenuMobile.js]: `begins` > `ends`.

File: cfgov/unprocessed/js/organisms/MegaMenuMobile.js [hunk 2]
```diff
@@ -173,21 +169,34 @@ function MegaMenuMobile( menus ) {
...
+   * Use this to perform post-expandBegin actions.
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/organisms/MegaMenuMobile.js]: `expandBegin` > `expandEnd`.

File: cfgov/unprocessed/js/organisms/MegaMenuMobile.js [hunk 3]
```diff
@@ -173,21 +169,34 @@ function MegaMenuMobile( menus ) {
...
+   * @param {Event} event - A FlyoutMenu event.
+   */
+  function _handleExpandEnd( event ) {
+    const menu = event.target;
+    const menuNode = menu.getData();
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/organisms/MegaMenuMobile.js]: Looks `menuNode` isn't used in this method and can be removed.

File: cfgov/unprocessed/js/organisms/MegaMenuMobile.js [hunk 4]
```diff
@@ -93,6 +95,7 @@ function MegaMenuMobile( menus ) {
     const menu = event.target;
     const rootMenu = _menus.getRoot().data;
     const menuNode = menu.getData();
+    const menuDom = menu.getDom();
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/organisms/MegaMenuMobile.js]: This isn't used is it?

Comment (jimmynotjim) [reply cfgov/unprocessed/js/organisms/MegaMenuMobile.js]: Nope, good catch

**Pull request merged**